File: src/test/resources/lambdas/SimpleLambda.java
package lambdas;

import java.util.function.Consumer;
import java.util.function.Function;
import java.util.function.Supplier;

public class SimpleLambda {
    private static final Supplier<String> supplier = () -> "supplier";
    private static final Consumer<String> consumer = s -> {
        System.out.println("consumer " + s);
    };
    private static final Function<String, String> function = s -> s + "+";

    public static void main(String[] args) {
        String a = supplier.get();
        consumer.accept("a");
        String b = function.apply("b");
    }
}

File: src/test/resources/lambdas/SimpleLambda.cs
namespace lambdas {
    public class SimpleLambda {
        private static readonly System.Func<string> supplier = () => "supplier";

        private static readonly System.Action<string> consumer = (string s) =>  {
            System.Console.Out.WriteLine("consumer " + s);
        };

        private static readonly System.Func<string, string> function = (string s) => s + "+";

        public static void Main(string[] args) {
            string a = supplier.Invoke();
            consumer.Invoke("a");
            string b = function.Invoke("b");
        }
    }
}